### PR TITLE
lesson progress

### PR DESCRIPTION
**Pull request by thepuskar**
Fixes #issue_number

### Description
- Tracking lesson progress one the basis of lesson test and progress
- Enabe next lesson

### CODE REVIEW

**Review by lmmrssa — CHANGES_REQUESTED** [review]
seems like lesson_progress is not being saved. also lesson 1 should only be active once user join the course.

**Review by lmmrssa — CHANGES_REQUESTED** [review]
@thepuskar seems like lesson progress is only saved when test is taken. it should save when course lesson is visited and mark as complete when test done
For lesson without test next button is not enabled

File: src/components/myCoursesCard/courseDescription/LessonCourseSingle.jsx [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
 
+  useEffect(() => {
+    setIsLessonCompleted(data?.lesson_progresses[0]?.isCompleted)
+    setUserId(data?.lesson_progresses[0]?.userId || null)
+  }, [data])
+
+  const lessonStart = async (id) => {
+    // console.log(id)
+    await setClickLessonId(id)
+    const addProgress = () => {
```
Comment (lmmrssa) [inline src/components/myCoursesCard/courseDescription/LessonCourseSingle.jsx:32]: seems like this is still WIP as addProgress is not being called so it does not seem to record or navigate to lesson

**Review by lmmrssa — CHANGES_REQUESTED** [review]

**Comment by lmmrssa** [comment]
@SujalLama process is still not working as expected. It is saving progress when course is viewed not on lesson

**Review by dogi — APPROVED** [review]
@SujalLama @thepuskar @lmmrssa congrats :P

**Review by lmmrssa — APPROVED** [review]

**Pull request merged**